### PR TITLE
Upgrade buildings

### PR DESCRIPTION
**Pull request by rzulian**
Create available commands for the different upgrades. Bescods peculiarities have been managed. 
I decided to create two different academy building, because they have different behaviours.
Still missing: payCosts, check if the mine is isolated, check if the planet is in the range

### CODE REVIEW

**Comment by rzulian** [comment]
@coyotte508 I tried, please have a look at the result. I definitely need a tutorial to understand how to deal with upstream / origin / brand and pull requests

**Comment by coyotte508** [comment]
@rzulian  One of the new tests is failing.

Regarding handling multiple remotes, you should have the `origin` remote pointing to your own repository. 

You can add a `main` remote pointing to this repository: `git remote add main https://github.com/donkeytech/gaia-engine`.

Then you have two possibilities:

- To fetch new changes, do `git pull main master`
- If you already committed locally, you can do `git pull main master --rebase`. If will unwind your local commits, apply the commits from the `main` remote, and then apply the local commits. If there are conflicts, the rebase will stop, you can edit the conflicting files and continue the rebase.

File: src/available-command.ts [hunk 1]
```diff
@@ -59,9 +59,6 @@ export function generate(engine: Engine): AvailableCommand[] {
   // We are in a regular round
   const commands = [];
   const player = engine.currentPlayer;
-
-  assert(player !== undefined, "Problem with the engine, player to play is unknown");
```
Comment (coyotte508) [inline src/available-command.ts]: If you remove the `assert` here, there's no need to keep it in the `import`

**Comment by coyotte508** [comment]
You can also edit an existing commit with `git commit --amend`, and forcefully overwrite commits with `git push force`.

Here is a cheat sheet: https://www.git-tower.com/blog/git-cheat-sheet/

**Pull request merged**